src/expr/src/relation/func.rs

```diff
@@ -138,17 +138,18 @@ fn sum_numeric<'a, I>(datums: I) -> Datum<'a>
 where
     I: IntoIterator<Item = Datum<'a>>,
 {
-    let datums = datums
-        .into_iter()
-        .filter(|d| !d.is_null())
-        .map(|d| d.unwrap_numeric().0)
-        .collect::<Vec<_>>();
-    if datums.is_empty() {
-        Datum::Null
-    } else {
-        let mut cx = numeric::cx_datum();
-        let sum = cx.sum(datums.iter());
-        Datum::from(sum)
+    let mut cx = numeric::cx_datum();
+    let mut sum = Numeric::zero();
+    let mut empty = true;
+    for d in datums {
+        empty = false;
+        if !d.is_null() {
+            cx.add(&mut sum, &d.unwrap_numeric().0);
+        }
```

Comment:
```suggestion
        if !d.is_null() {
            cx.add(&mut sum, &d.unwrap_numeric().0);
            empty = false;
        }
```

Otherwise the sum of `NULL`s would return zero, right?

Not entirely sure because tests are passing. So either there is a missing test or I'm missing something.

Reply:
Omg I can't believe I missed that 🙈. Looks like we didn't have test coverage. I added some, and checked that the tests failed on the original version of this PR and pass on the new version.